FF1Lib/Classes/ClassesData.cs

```diff
@@ -349,27 +349,39 @@ public void BlackKnight(Flags flags)
 			{
 
 				var whiteslotsK = _spellPermissions[Classes.Knight].Where(s => ((int)s % 8) < 4).ToList();
-			
 				var whitetoblackslotsK = whiteslotsK.Select(s => (s + 4)).ToList();
-
 				var knightslots = whitetoblackslotsK.ToList();
 				_spellPermissions.ClearPermissions(Classes.Knight);
 				_spellPermissions.AddPermissionsRange(knightslots.Select(x => (Classes.Knight, x)).ToList());
-
+				var KnightLvl4Charge = new List<byte> { 0b00000000, 0b00000000, 0b00000000, 0b00000000, 0b00000000, 0b00000000, 0b00000000, 0b00000000, 0b00000000,
+														0b00000000, 0b00000000, 0b00000000, 0b00000000, 0b00001111, 0b00000000, 0b00001111, 0b00000000, 0b00001111,
+														0b00000000, 0b00001111, 0b00000000, 0b00001111, 0b00000000, 0b00001111, 0b00000000, 0b00001111, 0b00000000,
+														0b00001111, 0b00000000, 0b00001111, 0b00000000, 0b00001111, 0b00000000, 0b00001111, 0b00000000, 0b00001111,
+														0b00000000, 0b00001111, 0b00000000, 0b00001111, 0b00000000, 0b00001111, 0b00000000, 0b00001111, 0b00000000,
+														0b00001111, 0b00000000, 0b00001111, 0b00000000 };
+				_classes[(int)Classes.Knight].SpCGrowth = KnightLvl4Charge;
```

Comment:
not a big deal, but since you do the assignment in both `if` brackets anyway, you could do it only once outside of them